app/models/message.rb

```diff
@@ -57,7 +57,8 @@ class Message < ApplicationRecord
     form: 6,
     article: 7,
     incoming_email: 8,
-    input_csat: 9
+    input_csat: 9,
+    location: 10
```

Comment:
@salman652 Can we add this as an attachment instead of saving this information inside the message object? Please see the attachment enum. There is already a location attribute. https://github.com/chatwoot/chatwoot/blob/develop/app/models/attachment.rb#L41

```
    {
      coordinates_lat: lat,
      coordinates_long: long,
      fallback_title: 'Address'
    }
```

On the rendering part in the UI, we can use the component you have created above.

Reply:
Hi @pranavrajs , i used attachment model for location as you mention in comment. On the rendering part in the UI side, i created separate component for location because previous location UI changes not fulfill the requirement for attachment model.